src/ophyd_async/plan_stubs/_ensure_connected.py

```diff
@@ -9,18 +11,19 @@ def ensure_connected(
     timeout: float = DEFAULT_TIMEOUT,
     force_reconnect=False,
 ):
-    (connect_task,) = yield from bps.wait_for(
-        [
-            lambda: wait_for_connection(
-                **{
-                    device.name: device.connect(
-                        mock=mock, timeout=timeout, force_reconnect=force_reconnect
-                    )
-                    for device in devices
-                }
+    if len(devices) != len({device.name for device in devices}):
+        raise RuntimeError(f"Devices do not have unique names {devices}")
```

Comment:
Could: Use something more specific than `RuntimeError`, maybe `ValueError` fits here?

Nit: This also fails if the user just passes the same device twice, I'm not too opinionated about whether we want to support that, but the error message makes less sense in that case.

Reply:
Additionally, having played with this, some sort of diff or other useful information for finding the conflict would be useful. If I have 30 devices and 1 duplicate name it may be hard to spot.

Reply:
@coretl as requested, can't think of a one-liner but this is close

```python
device_names = [device.name for device in devices]
duplicates = {name for name in device_names if device_names.count(name) > 1}
```